nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-jetty/src/main/java/org/apache/nifi/web/server/JettyServer.java

```diff
@@ -602,6 +599,28 @@ private WebAppContext loadWar(final File warFile, final String contextPath, fina
         return webappContext;
     }
 
+    private void addHTTPHeaders(WebAppContext webappContext) {
+        // Add a filter to set the X-Frame-Options header
+        FilterHolder xfoFilter = new FilterHolder(new XFrameOptionsFilter());
+        xfoFilter.setName(XFrameOptionsFilter.class.getSimpleName());
+        webappContext.addFilter(xfoFilter, "/*", EnumSet.allOf(DispatcherType.class));
+
+        // Add a filter to set the Content Security Policy frame-ancestors directive
+        FilterHolder cspFilter = new FilterHolder(new ContentSecurityPolicyFilter());
```

Comment:
This pattern seems like it could be abstracted to a single method which accepts a `Class<? extends Filter> clazz` and a `String path` and be called 4 times (iterate over a list and use a static path). This way future callers don't need to explicitly set the `DispatcherType` values, which was a problem before.

Reply:
I've updated the PR with a commit to do what I think you're suggesting. Let me know if it's in any way correct :)

Reply:
Yes, that did what I was describing. Thank you.